hermeto/core/package_managers/pip.py

```diff
@@ -2231,6 +2232,42 @@ def _version(dep: dict[str, Any]) -> str:
     }
 
 
+def _has_rust_build_deps(raw_build_dependencies: list[str]) -> bool:
+    rust_build_deps = ("maturin", "setuptools-rust", "setuptools_rust")
+    for dep in raw_build_dependencies:
+        if dep.strip().lower().startswith(rust_build_deps):
+            return True
+    return False
+
+
+def _depends_on_rust(source_tarball: tarfile.TarFile) -> bool:
+    file_parser_map = {
+        "pyproject.toml": parsers.parse_pyproject_toml,
+        "setup.cfg": parsers.parse_setup_cfg,
+        "setup.py": parsers.parse_setup_py,
+    }
+    for file_name, parser in file_parser_map.items():
+        pkg_name = source_tarball.getnames()[0].split("/")[0]
+        try:
+            file: tarfile.ExFileObject = source_tarball.extractfile(f"{pkg_name}/{file_name}")  # type: ignore[assignment]
```

Comment:
in order to not have mypy complaining below, I added this type annotation (I got the type using a debugger). 

not sure if this is the best approach (I've zero experience with mypy)